File: src/main/java/com/vaadin/demo/component/icons/IconFonts.java
package com.vaadin.demo.component.icons;

import com.vaadin.flow.router.Route;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.icon.FontIcon;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("icon-fonts")
public class IconFonts extends Div {

  public IconFonts() {
    HorizontalLayout layout = new HorizontalLayout();
    layout.setSpacing(true);
    layout.addClassName("items-center");

    // tag::snippet[]
    FontIcon codeBranchIcon = new FontIcon("fa", "fa-code-branch");
    FontIcon userIcon =  new FontIcon("fa", "fa-user");

    layout.add(codeBranchIcon, userIcon);
    // end::snippet[]
    add(layout);
  }

  public static class Exporter extends DemoExporter<IconFonts> { // hidden-source-line
  }
}

File: frontend/demo/component/icons/icon-fonts.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/horizontal-layout';
import '@vaadin/icon';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('icon-fonts')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <vaadin-horizontal-layout theme="spacing" style="align-items: center">
        <!-- tag::snippet[] -->
        <vaadin-icon icon-class="fa fa-code-branch"></vaadin-icon>
        <vaadin-icon icon-class="fa fa-user"></vaadin-icon>
        <!-- end::snippet[] -->
      </vaadin-horizontal-layout>
    `;
  }
}
